### PR TITLE
SOLR-17072: CLI package to print problem JSON Path

### PR DESCRIPTION
**Pull request by mkhludnev**
https://issues.apache.org/jira/browse/SOLR-17072

for current repository.json it yields: 

```
com.jayway.jsonpath.InvalidPathException: Error in JSON Path:$[config].[requestHandler].[HANDLER-PATH].[_packageinfo_].[version]
        at org.apache.solr.packagemanager.PackageManager.jsonPathResolve(PackageManager.java:866)
        at org.apache.solr.packagemanager.PackageManager.verify(PackageManager.java:829)
        at org.apache.solr.packagemanager.PackageManager.deployPackage(PackageManager.java:363)
        at org.apache.solr.packagemanager.PackageManager.deploy(PackageManager.java:943)
        at org.apache.solr.cli.PackageTool.runImpl(PackageTool.java:187)
        at org.apache.solr.cli.ToolBase.runTool(ToolBase.java:52)
        at org.apache.solr.cli.SolrCLI.main(SolrCLI.java:153)
Caused by: com.jayway.jsonpath.InvalidPathException: Could not parse token starting at position 1. Expected ?, ', 0-9, *
        at com.jayway.jsonpath.internal.path.PathCompiler.fail(PathCompiler.java:638)
        at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:139)
        at com.jayway.jsonpath.internal.path.PathCompiler.readContextToken(PathCompiler.java:123)
        at com.jayway.jsonpath.internal.path.PathCompiler.compile(PathCompiler.java:58)
        at com.jayway.jsonpath.internal.path.PathCompiler.compile(PathCompiler.java:75)
        at com.jayway.jsonpath.JsonPath.<init>(JsonPath.java:97)
        at com.jayway.jsonpath.JsonPath.compile(JsonPath.java:515)
        at com.jayway.jsonpath.internal.JsonContext.pathFromCache(JsonContext.java:222)
        at com.jayway.jsonpath.internal.JsonContext.read(JsonContext.java:77)
        at org.apache.solr.packagemanager.PackageManager.jsonPathResolve(PackageManager.java:864)
        ... 6 more

ERROR: Error in JSON Path:$[config].[requestHandler].[HANDLER-PATH].[_packageinfo_].[version]
```

### CODE REVIEW

**Review by epugh — APPROVED** [review]

**Comment by mkhludnev** [comment]
Thanks for approve @epugh. I'm thinking about test coverage. On top of my head only a simple unittest with mocks. WDYT?

File: solr/core/src/java/org/apache/solr/packagemanager/PackageManager.java [hunk 1]
```diff
@@ -1094,9 +1102,8 @@ public Map<String, String> getDeployedCollections(String packageName) {
...
       String version = null;
       try {
-        version =
-            JsonPath.parse(paramsJson, PackageUtils.jsonPathConfiguration())
-                .read("$['response'].['params'].['PKG_VERSIONS'].['" + packageName + "'])");
+        String jsonPath = "$['response'].['params'].['PKG_VERSIONS'].['" + packageName + "'])";
+        version = jsonPathResolve(paramsJson, jsonPath);
       } catch (PathNotFoundException ex) {
         // Don't worry if PKG_VERSION wasn't found. It just means this collection was never touched
         // by the package manager.
```
Comment (mkhludnev) [inline solr/core/src/java/org/apache/solr/packagemanager/PackageManager.java:1109]: or it's a redundant bracket in the end of json path

**Comment by mkhludnev** [comment]
oh gosh.. discovered one more error 
```
com.jayway.jsonpath.InvalidPathException: Error in JSON Path:$['response'].['params'].['PKG_VERSIONS'].['question-answer'])
	at org.apache.solr.packagemanager.PackageManager.jsonPathResolve(PackageManager.java:863)
	at org.apache.solr.packagemanager.PackageManager.getDeployedCollections(PackageManager.java:1106)
```
fixing right here

File: solr/modules/extraction/src/test/org/apache/solr/handler/extraction/TestXLSXResponseWriter.java [hunk 2]
```diff
@@ -38,8 +38,10 @@
 import org.apache.solr.search.SolrReturnFields;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
 import org.junit.Test;
 
+@Ignore
```
Comment (epugh) [inline solr/modules/extraction/src/test/org/apache/solr/handler/extraction/TestXLSXResponseWriter.java:44]: Is this meant to be here as part of the PR< or just to get past a build issue?

**Comment by epugh** [comment]
I love that you added a test.  Is the test as simple as it can be?   I think you are testing (in this test) the parsing of json paths, however I see some references to the V2 api and maybe some commented out code?    Is this test actually testing a packaget hat levers teh v2 api?     Also, I wonder if the setup/tearDown can be similified?

File: solr/modules/extraction/src/test/org/apache/solr/handler/extraction/TestXLSXResponseWriter.java (hunk 2 above)
Comment (mkhludnev) [reply solr/modules/extraction/src/test/org/apache/solr/handler/extraction/TestXLSXResponseWriter.java:44]: Just a build fix. Won't commit.

**Pull request merged**